src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,47 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_SIZE = 10;
+    private final V[] values;
+    private final K[] keys;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public StorageImpl() {
+        values = (V[]) new Object[STORAGE_SIZE];
+        keys = (K[]) new Object[STORAGE_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getKey(key);
+        if (index < 0) {
+            values[size] = value;
+            keys[size++] = key;
+        } else {
+            values[index] = value;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int index = getKey(key);
+        return index >= 0 ? values[getKey(key)] : null;
```

Comment:
```suggestion
        int index = getKeyIndex(key);
        return index != -1 ? values[index] : null;
```
you invoked getKeyIndex() twice